### DISCUSSION TITLE
Advice on overriding SingleRegion method

### DISCUSSION THREAD

**Discussion by bernardwiesner** [katspaugh/wavesurfer.js#3757] [Q&A]
Hi @katspaugh

I would like to override the RegionsPlugin, specifically the SingleRegion [avoidOverlapping](https://github.com/katspaugh/wavesurfer.js/blob/08afa79b8ead606ed732afb2acb020ab89ddf4a3/src/plugins/regions.ts#L467) method. 

My only thoughts of doing this is extending RegionsPlugin, and overriding the addRegion method to pass my implementation of the SingleRegion [here](https://github.com/katspaugh/wavesurfer.js/blob/08afa79b8ead606ed732afb2acb020ab89ddf4a3/src/plugins/regions.ts#L589C24-L589C36).

I would prefer not to override the addRegion method, just so I don't have to maintain it on upgrades on this package. Is there any other simpler way you can think of to customize the behavior of `avoidOverlapping` method ?

Thanks!

**Comment by katspaugh** [comment]
I'm guessing you just want to disable it? If so, please feel free to make a PR to make it optional in the Regions settings.

**Reply by bernardwiesner** [reply]
well, actually I would like to modify the behavior, but I guess I could do something similar to what was done here, passing some custom callback: https://github.com/katspaugh/wavesurfer.js/pull/3628/files

**Reply by katspaugh** [reply]
Ah, OK. Yes, a callback would be good too!
